src/platform/packages/shared/presentation/presentation_containers/interfaces/panel_management.ts

```diff
@@ -27,3 +28,19 @@ export interface CanExpandPanels {
 export const apiCanExpandPanels = (unknownApi: unknown | null): unknownApi is CanExpandPanels => {
   return Boolean((unknownApi as CanExpandPanels)?.expandPanel !== undefined);
 };
+
+export interface CanPinPanel {
+  pinPanel: (panelId: string) => void;
+  unpinPanel: (panelId: string) => void;
+  panelIsPinned: (panelId: string) => boolean;
+  addPinnedPanel: <StateType extends object, ApiType extends unknown = unknown>(
+    panel: PanelPackage<StateType>
+  ) => Promise<ApiType | undefined>;
+}
+
+export const apiCanPinPanel = (api: unknown): api is CanPinPanel => {
```

Comment:
Type guard should also check for `panelIsPinned` and `addPinnedPanel` since they are required by the interface.